contents/docs/self-host/configure/monitoring-with-grafana.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Monitoring with Grafana
+---
+
+This guide covers how to configure monitoring of your self-hosted deployment through Grafana.
+If you are targeting a production use-case, we _highly_ recommend setting up all of these options.
+
+## Getting started
+
+By default, the PostHog Helm chart does not come with Grafana enabled, so we will need to update our config values in order to install it.
+
+> **Note: ** This guide requires you to be running a Helm chart version of at least `26.0.6`. If you are running an older version, take a look at our guide on [how to upgrade PostHog](/docs/runbook/upgrading-posthog)
+
+## Setting up cluster monitoring
+
+This section covers setting up basic monitoring for the entire Kubernetes cluster, and provides basic metrics such as CPU usage, memory usage, and disk IOs.
+
+To set up basic monitoring, we will need to enable the following two charts:
+
+-   [grafana/grafana](https://github.com/grafana/helm-charts/tree/main/charts/grafana)
+-   [prometheus-community/prometheus](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus)
+
+which can be done by adding the following lines to our `values.yaml`
+
+```yaml
+grafana:
+    enabled: true
+prometheus:
+    enabled: true
+    alertmanager:
+        enabled: true
+```
+
+Next, we'll need to upgrade our deployment to spin-up the additional services, which can be done using the following command:
+
+```
+helm upgrade -f values.yaml --timeout 30m --namespace posthog posthog posthog/posthog --atomic --wait --wait-for-jobs --debug
+```
+
+### Connecting to Grafana
+
+Once our deployment is back up and running, we can now log in to Grafana to see our dashboards.
+By default, a single user is created with the username `admin` and an auto-generated password can be fetched by running:
+
+```
+kubectl -n posthog get secret posthog-grafana -o jsonpath="{.data.admin-password}" | base64 --decode
+```
+
+Next, we'll connect to Grafana by port-forwarding into the `posthog-grafana` service:
+
+```
+kubectl -n posthog port-forward svc/posthog-grafana 8080:80
+```
+
+Our Grafana dashboard should now be available at `localhost:8080`, and we can log in using the username `admin` along with the password we just retrieved.
+
```

Comment:
Note: users can also expose the service via an `Ingress` without having to use the port forwarding. By exposing the service they will be able to reach the service directly using a DNS name (or IP)